gr-blocks/lib/selector_impl.cc

```diff
@@ -78,6 +83,37 @@ void selector_impl::set_output_index(unsigned int output_index)
         throw std::out_of_range("output_index must be < noutputs");
 }
 
+void selector_impl::handle_msg_input_index(pmt::pmt_t msg)
+{
+    pmt::pmt_t data = pmt::cdr(msg);
+
+    if (pmt::is_integer(data)) {
+        const unsigned int new_port = pmt::to_long(data);
+
+        if (new_port < (d_num_inputs)) {
```

Comment:
Minor: Those parens are redundant, and you might want to add an else case with logging, but nothing that would hold up a merge.

Reply:
Please -do- add an "else" clause here with logging warnings about the integer value being out of range. Same thing in the code below. When there's something not happening that could be, we should warn the user such that they might be able to fix the issue.